### PR TITLE
refactor(model): get dataset

### PR DESCRIPTION
**Pull request by boss-chanon**
## Why this PR
refactor get_dataset to use config instead args

## Changes
- get_dataset to use config instead args
- make yaml file for dataset

## Related Issues
Close #

## Checklist
- [ ] PR should be in the [Naming convention](../../docs/PR_NAMING.md)
- [ ] Assign yourself in to Assigneees
- [ ] Tag related issues
- [ ] Constants name should be ALL_CAPITAL, function name should be snake_case, and class name should be CamelCase
- [ ] complex function/algorithm should have [Docstring](https://peps.python.org/pep-0257/)
- [ ] 1 PR should not have more than 200 lines changes (Exception for test files). If more than that please open multiple PRs
- [ ] At least PR reviewer must come from the task's team (model, eval, data)

### CODE REVIEW

File: src/model/configuration_example/model/falcon-7b-from-pretrained.yaml [hunk 1]
```diff
@@ -17,3 +17,18 @@ args:
 tokenizer:
   pretrained_model_name_or_path: tiiuae/falcon-7b
   tokenizer_class: AutoTokenizer
+tokenized_dataset: None
```
Comment (boat1603) [inline src/model/configuration_example/model/falcon-7b-from-pretrained.yaml:20]: We should set the dataset config in `example_config.yaml` file.

File: src/model/openthaigpt_pretraining_model/datasets/__init__.py [hunk 2]
```diff
@@ -3,7 +3,7 @@
 from typing import Union
 
 
-def get_dataset(
+def get_dataset_for_lion(
```
Comment (boat1603) [inline src/model/openthaigpt_pretraining_model/datasets/__init__.py:6]: You should copy this whole function into lion experiment scrpits (Note: We will refactor it later.)

File: src/model/configuration_example/model/falcon-7b-from-pretrained.yaml (hunk 1 above)
Comment (boss-chanon) [reply src/model/configuration_example/model/falcon-7b-from-pretrained.yaml]: done

File: src/model/openthaigpt_pretraining_model/datasets/__init__.py (hunk 2 above)
Comment (boss-chanon) [reply src/model/openthaigpt_pretraining_model/datasets/__init__.py]: done

**Review by new5558 — APPROVED** [review]

**Review by boat1603 — APPROVED** [review]

**Pull request merged**